src/js/media-controller.js

```diff
@@ -325,6 +347,21 @@ class MediaController extends MediaContainer {
         }
         this.propagateMediaState(MediaUIAttributes.MEDIA_IS_PIP, isPip);
       },
+      // Note this relies on a customized video[is=castable-video] element.
+      'entercast,leavecast': (e) => {
+        const isCast = this.media && globalThis.CastableVideo?.castElement === this.media;
+
+        // Prevent auto hiding controls while casting.
+        if (isCast && this.autohide >= 0) {
+          this._prevAutohide = this.autohide;
+          this.setAttribute('autohide', '-1');
+          this.removeAttribute('user-inactive');
+        } else if (this._prevAutohide >= 0) {
+          this.setAttribute('autohide', this._prevAutohide);
+        }
+
+        this.propagateMediaState(MediaUIAttributes.MEDIA_IS_CAST, isCast);
```

Comment:
non-blocker: consider refactoring this to handle multiple well-defined states instead of on/off to support "fancier" UIs in the future per Google design recommendations (see e.g. https://developers.google.com/cast/docs/design_checklist/cast-button#sender-cast-icon-states).

Reply:
I added the "connecting" value to media-is-cast for media state receivers and added the original Google svg's with classes for the wifi arches in the icon. only in my testing this connecting state only took a split second. I tested the <google-cast-launcher> button and it also didn't show the animation on my end. maybe my wifi is too good :P 

I'll look into animating the SVG today, in a 3rd icon <slot> I imagine for it to be replaceable